### PR TITLE
Endpoint to fetch jobs discovery

### LINKED ISSUE #10: Jobs Discovery
The backend application will receive selected Exchange Oracle address in the request, then query _GET [EXCHANGE_ORACLE_URL]/job_ for available jobs and return them as a response. This won’t involve any caching for now. This solution should support all fields mentioned in the tech-docs.

Acceptance criteria:
- The backend must accept requests containing a selected Exchange Oracle address. This address should be used to construct the target URL for querying available jobs (e.g., _[EXCHANGE_ORACLE_URL]/job_).
- The request to the Exchange Oracle can include all fields mentioned in the tech-docs.

Notes:
- Discuss possibility of validating the EXCHANGE_ORACLE_URL with Redis cache to avoid requesting non-existing oracles

### PR DESCRIPTION
**Pull request by MWBlocky**
## Description

This Pull Request adds the jobs discovery endpoint.

## Summary of changes

- Added job discovery module.
- Added unit tests.

## Related issues
Fixes #10

### CODE REVIEW

File: packages/apps/human-app/server/src/modules/jobs-discovery/jobs-discovery.service.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    );
+    try {
+      const options = {
+        method: 'GET',
+        url: `${url}/jobs`,
+        params: jobsDiscoveryParamsData,
+      };
+      const response = await lastValueFrom(this.httpService.request(options));
+      return response.data;
+    } catch (error) {
```
Comment (macnablocky) [inline packages/apps/human-app/server/src/modules/jobs-discovery/jobs-discovery.service.ts:36]: Didn't we agreed that this kind of error cases would be handled by exception filters?

File: packages/apps/human-app/server/src/modules/jobs-discovery/jobs-discovery.service.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+      JobsDiscoveryParamsCommand,
+      JobsDiscoveryParamsData,
+    );
+    try {
+      const options = {
+        method: 'GET',
+        url: `${url}/jobs`,
+        params: jobsDiscoveryParamsData,
+      };
+      const response = await lastValueFrom(this.httpService.request(options));
```
Comment (macnablocky) [inline packages/apps/human-app/server/src/modules/jobs-discovery/jobs-discovery.service.ts:34]: what is the motivation behind lastValueFrom

**Review by macnablocky — APPROVED** [review]

File: packages/apps/human-app/server/src/modules/jobs-discovery/jobs-discovery.service.ts (hunk 1 above)
Comment (MWBlocky) [reply packages/apps/human-app/server/src/modules/jobs-discovery/jobs-discovery.service.ts:36]: It's will be done in the next step -> https://github.com/orgs/blockydevs/projects/1/views/1?pane=issue&itemId=54595048

File: packages/apps/human-app/server/src/modules/jobs-discovery/jobs-discovery.service.ts (hunk 2 above)
Comment (MWBlocky) [reply packages/apps/human-app/server/src/modules/jobs-discovery/jobs-discovery.service.ts:34]: HttpService methods is an Observable -> [docs](https://docs.nestjs.com/techniques/http-module)

**Pull request merged**